### DISCUSSION TITLE
Stats API - add a more flexible filter for page paths

### DISCUSSION THREAD

**Discussion by RobertJoonas** [plausible/analytics#1282] [Anything Else]
With Stats API, It's currently not possible to pull data from all pages that start with, say "/article/". A simple "that_contains" option would do the trick, but a regex filter would be the best.

(feedback via email)

**UPDATE:** This is now live!

**Comment by hadley** [comment]
Or just support the same syntax that the app uses for filtering, e.g. `{"page":"/blog/**"}`

**Comment by luctus** [comment]
Hello @metmarkosaric,

Do we have any updates on this? thanks!

**Reply by metmarkosaric** [reply]
Hey @luctus! There are no updates on this and no short term plans that I can tell you about. We are focused on the more frequently requested features such as export/import at the moment. We'd like to build this but we are a small team with limited development resources so need to focus on most frequently requested features one at a time. Thanks for understanding!

**Reply by luctus** [reply]
Totally understandable. Thanks for the fast reply!

cheers!

**Comment by wladpaiva** [comment]
This seems like a really low priority for you guys. Could we have at least some directions on where this should be implemented in case we open a PR?

**Reply by mquandalle** [reply]
It seems to work : `?event:page==/path/**`

**Comment by RobertJoonas** [comment]
Apologies everyone! It seems like we've missed updating this thread. I can see this was added 8 months ago by @ukutaht in https://github.com/plausible/analytics/commit/0a5f6c5886bd2c827c77d7541f183430d276a576.

I can confirm that the Stats API does support the same wildcard filtering that we have in the dashboard.

**Reply by metmarkosaric** [reply]
oh wow! thanks for checking! i've been telling people it's not live yet 😱

**Reply by metmarkosaric** [reply]
@RobertJoonas could you explain this use case in the stats API docs please? you can use some of the same info that we have here: https://plausible.io/docs/pageview-goals#pageview-goals-support-wildcards. thanks!

**Comment by wladpaiva** [comment]
Weird. This doesn't:
```
{{BASE_URL}}/api/v1/stats/aggregate?site_id=eatingdots.com&filters=event:page==/bolos-de-julieta**
```

I'm using the docker version `plausible/analytics:latest` updated 5 days ago so I'm assuming I've got the correct version

**Reply by luctus** [reply]
I understand these wildcards only work when creating page goals, not in the stats api, sadly.

**Reply by luctus** [reply]
I just tested it in production and it worked! I wasn't aware of it, I will update my app right now.

Anyway, you need to encode that url, so it should be something like this:

```
{{BASE_URL}}/api/v1/stats/aggregate?site_id=eatingdots.com&filters=event%3Apage%3D%3D%2Fbolos-de-julieta**
```

**Reply by wladpaiva** [reply]
I could definitely be doing something wrong but it looks like it only works for the web app but not for the API. It didn't like the encoded version version though. Are you running on the most recent version?

![Screen Shot 2022-09-14 at 5 59 40 PM](https://user-images.githubusercontent.com/1843792/190261512-ac281474-bd89-40a5-86d3-43afa2c0e319.png)
![Screen Shot 2022-09-14 at 5 59 18 PM](https://user-images.githubusercontent.com/1843792/190261526-005f1b7a-91bd-4b55-b078-9515bc731907.png)

**Reply by luctus** [reply]
That's weird. You are using the cloud version, right? If you want, you can test it in dailytics.com and I can see the logs to see the exact query string

**Reply by wladpaiva** [reply]
I'm using the Self-hosted. Good idea!

**Comment by RobertJoonas** [comment]
Clarifying here, that this feature did not make it to the latest self-hosted release. At this stage, wildcards are only supported on the official cloud-hosted version.

Self-hosted is not updated as frequently as the hosted version which changes daily so the latest features won’t be immediately available. Think of it as a long-term release - we stack all the new features and fixes that have been battle tested on the hosted version. We'd like to release a new self-hosted version soon but cannot make any strict promises. Thanks for understanding!

**Reply by wladpaiva** [reply]
@RobertJoonas Totally understandable. It just doesn't make sense because if the feature wasn't available already in the self-hosted I shouldn't be able to use it on the web app. Right?
This is not the case, the feature works 100% on the self-hosted web app. Shouldn't the web app be using the same API as I do?

**Reply by RobertJoonas** [reply]
@wladiston
> It just doesn't make sense because if the feature wasn't available already in the self-hosted I shouldn't be able to use it on the web app. Right?

Not quite. Dashboard wildcard filtering is supported from [version 1.4.0](https://github.com/plausible/analytics/releases/tag/v1.4.0) (Oct 27, 2021). Filtering in Stats API works somewhat differently than the dashboard, and wildcard filtering wasn't included in the API at that time.

So in the current self-hosted release, you can use wildcards in the dashboard, but not in Stats API. Hope this makes sense.

**Reply by polman4** [reply]
Hello, any update for this?

**Comment by danielegaliffa** [comment]
Hello, is there a possibility to use a regular expression filter with the STATs API to select pages that do not contain a specific string?

Thanks
